api/src/Entity/Period.php

```diff
@@ -113,6 +115,10 @@ class Period extends BaseEntity implements BelongsToCampInterface {
      */
     #[Assert\LessThanOrEqual(propertyPath: 'end')]
     #[ApiProperty(example: '2022-01-01', openapiContext: ['format' => 'date'])]
+    #[Context(
+        normalizationContext: [DateTimeNormalizer::FORMAT_KEY => 'Y-m-d'],
+        denormalizationContext: [DateTimeNormalizer::FORMAT_KEY => '!Y-m-d']
```

Comment:
Was macht das Ausrufezeichen?

Reply:
Quote from https://www.php.net/manual/de/datetime.createfromformat.php:
> Resets all fields (year, month, day, hour, minute, second, fraction and timezone information) to zero-like values ( 0 for hour, minute, second and fraction, 1 for month and day, 1970 for year and UTC for timezone information) 

> Without !, all fields will be set to the current date and time.

D.h. DateTime::createFromFormat('Y-m-d', '2021-07-01') liefert ein DateTime object mit aktueller Zeit + Zeitzone des Servers.

Leider kann DateTime::format mit dem Ausrufezeichen nicht umgehen, bzw. druckt es effektiv als Ausrufezeichen aus. Deshalb kann nicht dasselbe Format verwendet werden.

Ziemlich komischer Design-Entscheid, dass so zu implementieren. But that's how it is...